save notes to local storage

chayboats opened the pull request:
Add 2 use effect hooks
One to get localNotes
One to set localNotes

chayboats committed:
save notes to local storage

chayboats committed:
#12: Fix styling for action icon

wdonray reviewed (approved):
![istockphoto-652331292-612x612](https://github.com/chayboats/keeper/assets/21087934/5d34cd51-9dc6-4238-bbae-6e51efb53c51)

chayboats committed:
Create Account Page
Create a create account page
Add styles
Create reusable InvalidAlert component
Create LogIn, Logo, and SignUpButtons components to reduce code in index.js
Login Page
Move return under functions
Update useEffect Hook
The useEffect will now only fetch and set the user data in localStorge if the key does not Exist
If it does, we will get the data and set it to users

wdonray commented on pages/create-account/index.js:
```diff
@@ -0,0 +1,117 @@
+    passwordCheck() ? updateUsers() : setHidePasswordAlert(false);
+  }
+
+  function emailCheck() {
```
**suggestion: ** needs better name

wdonray commented on pages/create-account/index.js:
```diff
@@ -0,0 +1,117 @@
+    return Boolean(findEmail);
+  }
+
+  function passwordCheck() {
```
**suggestion: ** needs better name

wdonray commented on pages/create-account/index.js:
```diff
@@ -0,0 +1,117 @@
+  }
+
+  function emailCheck() {
+    const findEmail = users.find((user) => user.email == email);
```
emailFound

wdonray commented on pages/create-account/index.js:
```diff
@@ -0,0 +1,117 @@
+    return Boolean(match);
+  }
+
+  function updateUsers() {
```
**suggestion:** think of a better name

wdonray commented on pages/create-account/index.js:
```diff
@@ -0,0 +1,117 @@
+            message="* Your password must be 8-20 characters long, contain letters and numbers, and must not contain spaces, special characters, or emoji."
+          />
+          <SignUpButtons />
+          <LogIn />
```
**suggestion: ** This feels like it could use a better name...

chayboats committed:
Address naming conventions
